src/main/java/com/exasol/adapter/dialects/elasticsearch/ElasticSearchColumnMetadataReader.java

```diff
@@ -0,0 +1,31 @@
+package com.exasol.adapter.dialects.elasticsearch;
+
+import java.sql.Connection;
+
+import com.exasol.adapter.AdapterProperties;
+import com.exasol.adapter.dialects.IdentifierConverter;
+import com.exasol.adapter.jdbc.BaseColumnMetadataReader;
+
+/**
+ * This class implements ElasticSearch-specific reading of column metadata.
+ */
+public class ElasticSearchColumnMetadataReader extends BaseColumnMetadataReader {
+
+    /**
+     * Create a new instance of the {@link ElasticSearchColumnMetadataReader}.
+     *
+     * @param connection          JDBC connection through which the column metadata is read from the remote database
+     * @param properties          user-defined adapter properties
+     * @param identifierConverter converter between ElasticSearch and Exasol identifiers
+     */
+    public ElasticSearchColumnMetadataReader(final Connection connection, final AdapterProperties properties,
+            final IdentifierConverter identifierConverter) {
+        super(connection, properties, identifierConverter);
+    }
+
+    @Override
+    protected String mapColumnName(final String columnName) {
+        return ColumnNameMapper.mapToExasolDialect(columnName);
```

Comment:
In this particular case I think having the mapping functions here would have been OK.
You probably extracted them because of level of abstraction. I can live with that.

Reply:
Yes. But in particular I extracted it to have it both ways (Exasol-ElasticSearch and ElasticSearch-Exasol) in the same place cause they depend on each other, and as `ElasticSearchColumnMetadataReader.java` only uses one way.